http/src/request/channel/webhook/execute_webhook.rs

```diff
@@ -203,20 +199,20 @@ impl<'a> ExecuteWebhook<'a> {
         self
     }
 
-    /// If true, wait for the message to send before sending a response. See [Discord Docs/Execute
-    /// Webhook]
+    /// Wait for the message to send before sending a response. See
+    /// [Discord Docs/Execute Webhook].
+    ///
+    /// Using this will result in receiving the created message.
     ///
     /// [Discord Docs/Execute Webhook]: https://discord.com/developers/docs/resources/webhook#execute-webhook-querystring-params
-    pub fn wait(mut self, wait: bool) -> Self {
-        self.fields.wait.replace(wait);
-
-        self
+    pub fn wait(self) -> ExecuteWebhookAndWait<'a> {
```

Comment:
No; `start` and the `Future` implementation for `ExecuteWebhook` needs to return a concrete type. This request returns `Option<Message>` prior to this PR to handle 404s, but due to `Response::status` being accessible to handle 404s requests' `Future` outputs shouldn't resolve to a `Response<Option<T>>` anymore. By having a separate `ExecuteWebhookAndWait` request it can be typed as `Response<Message>` while `ExecuteWebhook` can resolve to a `Response<EmptyBody>`.